apps/exponent-liquidator/src/ExponentLiquidator.ts

```diff
@@ -0,0 +1,851 @@
+  Network,
+  sendTxThroughRelayer,
+} from '@notional-finance/util';
+import {
+  RiskyPosition,
+  EnrichedPosition,
+  Env,
+  Position,
+  TokenPrice,
+  VaultType,
+  TransactionResult,
+  LiquidationReport,
+  LiquidationRunResult,
+} from './types';
+import { MorphoRouterIntegration } from './utils/morphoRouter';
+import { getWithdrawRequestData } from './utils/withdrawRequestData';
+import { VaultRegistry } from './utils/vaultRegistry';
+import {
+  ExponentFlashLiquidatorV2,
+  ExponentFlashLiquidatorV2ABI,
+  TradingModule,
+  TradingModuleABI,
+} from '@notional-finance/contracts';
+import { generateRedeemData } from './utils/redeemDataGenerator';
+import { getTokenPrices } from './utils/tokenPricing';
+import { LiquidatorLogger } from './utils/logging';
+import { aggregate, AggregateCall } from '@notional-finance/multicall';
+import { VAULT_ABI } from './abis';
+
+// Math constants for liquidation calculations
+const ORACLE_PRICE_SCALE = ethers.utils.parseUnits('1', 36);
+
+const wMulDown = (x: ethers.BigNumber, y: ethers.BigNumber): ethers.BigNumber => {
+  const WAD = ethers.utils.parseUnits('1', 18);
+  return x.mul(y).div(WAD);
+};
+
+const mulDivDown = (x: ethers.BigNumber, y: ethers.BigNumber, d: ethers.BigNumber): ethers.BigNumber => {
+  return x.mul(y).div(d);
+};
+
+export default class ExponentLiquidator {
+  private provider: ethers.providers.Provider;
+  private morphoRouterIntegration: MorphoRouterIntegration;
+  private logger: LiquidatorLogger;
+  private vaultRegistry: VaultRegistry;
+  private positions: Position[];
+  private flashLiquidator: ExponentFlashLiquidatorV2;
+  private tradingModule: TradingModule;
+  private network: Network;
+  private environment: 'development' | 'production';
+
+  constructor(
+    private env: Env,
+    positions: Position[],
+    vaultRegistry: VaultRegistry,
+    provider: ethers.providers.Provider,
+    environment: 'development' | 'production' = 'production'
+  ) {
+    this.network = env.NETWORK;
+    this.provider = provider;
+    this.positions = positions;
+    this.vaultRegistry = vaultRegistry;
+    this.environment = environment;
+    this.morphoRouterIntegration = new MorphoRouterIntegration(
+      this.provider,
+      env.MORPHO_LENDING_ROUTER_ADDRESS
+    );
+    this.logger = new LiquidatorLogger(env);
+    this.flashLiquidator = new ethers.Contract(
+      env.FLASH_LIQUIDATOR_ADDRESS,
+      ExponentFlashLiquidatorV2ABI,
+      this.provider
+    ) as ExponentFlashLiquidatorV2;
+    this.tradingModule = new ethers.Contract(
+      env.TRADING_MODULE_ADDRESS,
+      TradingModuleABI,
+      this.provider
+    ) as TradingModule;
+  }
+
+  async getRiskyPositions(): Promise<RiskyPosition[]> {
+    // Step 1: Batch healthFactor calls to get raw health factor data
+    const healthFactorData =
+      await this.morphoRouterIntegration.batchHealthFactors(this.positions);
+
+    // Step 2: Batch borrowShares calls to get borrow share data
+    const borrowSharesData =
```

Comment:
You don't need this data here, you only need it for the accounts that have risky positions. I would not execute for all accounts here.